metashare/api/jobs.py

```diff
@@ -477,46 +478,48 @@ def populate_github_users(repository):
 def submit_review(*, user, task, data):
     try:
         review_sha = None
+        org = data["org"]
         task.refresh_from_db()
-        if task.commits and task.pr_is_open:
-            repo_id = task.project.repository.get_repo_id(user)
-            repository = get_repo_info(user, repo_id=repo_id)
-            pr = repository.pull_request(task.pr_number)
-
-            if data.get("org"):
-                review_sha = data["org"].latest_commit
-            elif task.review_valid:
-                review_sha = task.review_sha
-            else:
-                review_sha = task.commits[0].get("id")
-
-            # The values in this dict are the valid values for the
-            # `state` arg to repository.create_status. We are not
-            # currently using all of them, because some of them make no
-            # sense for our system to add to GitHub.
-            state_for_status = {
-                # "": "pending",
-                # "": "error",
-                TASK_REVIEW_STATUS.Approved: "success",
-                TASK_REVIEW_STATUS["Changes requested"]: "failure",
-            }.get(data["status"])
-
-            target_url = get_user_facing_url(
-                path=[
-                    "repositories",
-                    task.project.repository.slug,
-                    task.project.slug,
-                    task.slug,
-                ]
-            )
+        if org:
+            review_sha = org.latest_commit
+        elif task.review_valid:
+            review_sha = task.review_sha
 
-            repository.create_status(
-                review_sha,
-                state_for_status,
-                target_url=target_url,
-                description=data["notes"],
-                context="MetaShare Review",
-            )
+        if not (task.pr_is_open and review_sha):
+            raise ValidationError(_("Cannot submit review for this task."))
```

Comment:
`ValidationError` seems odd; it's particularly for use in DRF serializers. I will make a small change here for that.